File: JavaScript/20-Valid Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const hmap = {'(':')','{':'}','[':']'};
    stack = [];
    
    for (let char of s){
        if (stack.length>0 && hmap[stack[stack.length-1]] === char){
            stack.pop();
        }
        else{
            stack.push(char);
        }
    }
    return stack.length === 0;
};

File: 20-Valid Parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        
        hmap = {"(":")", "[":"]", "{":"}"}
        
        stack = []
        for char in s:
            if char in hmap:
                stack.append(char)
            elif stack and char == hmap[stack[-1]]:
                stack.pop()
            else:
                return False
        
        return not stack